passes/sat/qbfsat.cc

```diff
@@ -28,9 +28,16 @@
 USING_YOSYS_NAMESPACE
 PRIVATE_NAMESPACE_BEGIN
 
+static inline unsigned int difference(unsigned int a, unsigned int b) {
+	if (a < b)
+		return b - a;
+	else
+		return a - b;
+}
+
```

Comment:
Why not just `abs(a-b)`?

Reply:
Because `a` and `b` are `unsigned int`s.  If `b` > `a`, then `a - b` underflows (or maybe is undefined?  I am not familiar with the spec in this regard).  One option is to make `a` and/or `b` (in this case `success` and `failure` on line 470) signed `int`s, but that limits the maximum value that  `a` and `b` can have.  Another option is to use a bigger signed type, like `long long`, but that's not really the point.  I think a simple difference function is elegant, functionally obvious, has no signedness issues, and allows the use of the full `unsigned` value range.

FWIW using `abs(success - failure)` with `unsigned int`s on line 470 gives this compiler error:
```
passes/sat/qbfsat.cc: In function ‘{anonymous}::QbfSolutionType {anonymous}::qbf_solve(Yosys::RTLIL::Module*, const {anonymous}::QbfSolveOptions&)’:
passes/sat/qbfsat.cc:479:47: error: call of overloaded ‘abs(unsigned int)’ is ambiguous
   while (failure == 0 || abs(success - failure) > 1) {
                                               ^
In file included from /usr/include/c++/8/cstdlib:75,
                 from /usr/include/c++/8/ext/string_conversions.h:41,
                 from /usr/include/c++/8/bits/basic_string.h:6400,
                 from /usr/include/c++/8/string:52,
                 from /usr/include/c++/8/stdexcept:39,
                 from /usr/include/c++/8/array:39,
                 from /usr/include/c++/8/tuple:39,
                 from /usr/include/c++/8/bits/stl_map.h:63,
                 from /usr/include/c++/8/map:61,
                 from ./kernel/yosys.h:42,
                 from passes/sat/qbfsat.cc:20:
/usr/include/stdlib.h:837:12: note: candidate: ‘int abs(int)’
 extern int abs (int __x) __THROW __attribute__ ((__const__)) __wur;
            ^~~
In file included from /usr/include/c++/8/cstdlib:77,
                 from /usr/include/c++/8/ext/string_conversions.h:41,
                 from /usr/include/c++/8/bits/basic_string.h:6400,
                 from /usr/include/c++/8/string:52,
                 from /usr/include/c++/8/stdexcept:39,
                 from /usr/include/c++/8/array:39,
                 from /usr/include/c++/8/tuple:39,
                 from /usr/include/c++/8/bits/stl_map.h:63,
                 from /usr/include/c++/8/map:61,
                 from ./kernel/yosys.h:42,
                 from passes/sat/qbfsat.cc:20:
/usr/include/c++/8/bits/std_abs.h:78:3: note: candidate: ‘constexpr long double std::abs(long double)’
   abs(long double __x)
   ^~~
/usr/include/c++/8/bits/std_abs.h:74:3: note: candidate: ‘constexpr float std::abs(float)’
   abs(float __x)
   ^~~
/usr/include/c++/8/bits/std_abs.h:70:3: note: candidate: ‘constexpr double std::abs(double)’
   abs(double __x)
   ^~~
/usr/include/c++/8/bits/std_abs.h:61:3: note: candidate: ‘long long int std::abs(long long int)’
   abs(long long __x) { return __builtin_llabs (__x); }
   ^~~
/usr/include/c++/8/bits/std_abs.h:56:3: note: candidate: ‘long int std::abs(long int)’
   abs(long __i) { return __builtin_labs(__i); }
   ^~~
make: *** [Makefile:673: passes/sat/qbfsat.o] Error 1
```